File: SNS.Web/src/app/jobs/job-applications/contracts/job-application-summary.dto.ts
import { ApplicationStatus } from '../../enums/application-status.enum';

export interface JobApplicationSummaryDto {
    id: string;
    jobId: string;
    jobTitle: string;
    companyName: string;
    applicantId: string;
    applicantFullName: string;
    status: ApplicationStatus;
    createdAt: string;
}

File: backend/SNS.Application/Jobs/JobApplications/Contracts/JobApplicationSummaryDto.cs
using SNS.Domain.QA.Enums;

namespace SNS.Application.Jobs.JobApplications.Contracts;

public sealed record JobApplicationSummaryDto(
    Guid Id,
    Guid JobId,
    string JobTitle,
    string CompanyName,
    Guid ApplicantId,
    string ApplicantFullName,
    string? ApplicantAvatarUrl,
    string? ApplicantSpecialization,
    Guid? ResumeId,
    string? ResumeFileUrl,
    ApplicationStatus Status,
    DateTime CreatedAt,
    DateTime UpdatedAt
);
